stdparams.go

```diff
@@ -0,0 +1,38 @@
+// +build go1.7
+
+package httprouter
+
+import (
+	"context"
+	"net/http"
+)
+
+type paramsKey struct{}
```

Comment:
As I already pointed out I am not questioning the existence of a `paramsKey` type in general but I propose to define this just like [the canonical example in the context package documentation does][1]:

```go
// contextKey is an unexported type for context keys defined in this package.
// This prevents collisions with context keys defined in other packages.
type contextKey int
```

This will always prevent collisions with context keys from other packages since the hash keys are just of type `interface{}` and comparison on those is defined to take the actual type into account:

From the [Go Language Specification][2]
> A value x of non-interface type X and a value t of interface type T are comparable when values of type X are comparable and X implements T. They are equal if t's dynamic type is identical to X and t's dynamic value is equal to x.

I prepared this example to demonstrate this useful property: https://play.golang.org/p/T1UYUbmonh

[1]: https://godoc.org/context#Context.Value
[2]: https://golang.org/ref/spec#Comparison_operators